### PR TITLE
Handle nil stripe customer in credit card clone removal logic

### PR DESCRIPTION
**Pull request by arku**
#### What? Why?

Fixes the code not checking if the customer exists before calling `deleted?`.

#### Release notes

Changelog Category: User facing changes | Technical changes

### CODE REVIEW

File: lib/stripe/credit_card_clone_destroyer.rb [hunk 1]
```diff
@@ -17,7 +17,9 @@ def destroy_clones(card)
         next unless customer_id
 
         customer = Stripe::Customer.retrieve(customer_id, stripe_account: stripe_account)
-        customer&.delete unless customer.deleted?
+        next unless customer
+
+        customer.delete unless customer.deleted?
```
Comment (luisramos0) [inline lib/stripe/credit_card_clone_destroyer.rb:22]: wouldnt this do the trick?
        `customer.delete if customer && !customer.deleted?`

File: lib/stripe/credit_card_clone_destroyer.rb [hunk 2]
```diff
@@ -17,7 +17,9 @@ def destroy_clones(card)
         next unless customer_id
 
         customer = Stripe::Customer.retrieve(customer_id, stripe_account: stripe_account)
```
Comment (luisramos0) [inline lib/stripe/credit_card_clone_destroyer.rb:19]: we are reusing variables here :see_no_evil: 
customer is a ofn customer above and here we becomes a stripe customer. this variable should be named stripe_customer, not customer. can you please improve this? Thanks!

Comment (andrewpbrett) [reply lib/stripe/credit_card_clone_destroyer.rb]: 🙈 😲 👍

**Review by andrewpbrett — APPROVED** [review]
Thanks @arku and @luisramos0 for spotting that variable reuse!

**Comment by andrewpbrett** [comment]
I'm going to merge since the feedback from @luisramos0 in his review has been addressed.

**Pull request merged**